atomicapp/nulecule/lib.py

```diff
@@ -64,46 +63,21 @@ def load_config(self, config, ask=False, skip_asking=False):
             None
         """
         for param in self.params:
-            value = config.get(self.namespace, {}).get(param[NAME_KEY]) or \
-                config.get(GLOBAL_CONF, {}).get(param[NAME_KEY])
+            value = config.get(param[NAME_KEY], scope=self.namespace)
             if value is None and (ask or (
                     not skip_asking and param.get(DEFAULTNAME_KEY) is None)):
                 cockpit_logger.info("%s is missing in answers.conf." % param[NAME_KEY])
                 value = Utils.askFor(param[NAME_KEY], param)
             elif value is None:
                 value = param.get(DEFAULTNAME_KEY)
-            if config.get(self.namespace) is None:
-                config[self.namespace] = {}
-            config[self.namespace][param[NAME_KEY]] = value
+            config.set(param[NAME_KEY], value, scope=self.namespace, source='runtime')
         self.config = config
 
-    def merge_config(self, to_config, from_config):
-        """
-        Merge values from from_config to to_config. If value for a key
-        in a group in to_config is missing, then only set it's value from
-        corresponding key in the same group in from_config.
-
-        Args:
-            to_config (dict): Dictionary to merge config into
-            from_config (dict): Dictionary to merge config from
-
-        Returns:
-            None
-        """
-        for group, group_vars in from_config.items():
-            to_config[group] = to_config.get(group) or {}
-            for key, value in (group_vars or {}).items():
-                if to_config[group].get(key) is None:
-                    to_config[group][key] = value
-
     def get_context(self):
         """
         Get context data from config data for rendering an artifact.
```

Comment:
We can just get rid of this method with ``context = self.config.context(self.namespace)``